ocaml/xapi/xapi_pool.ml

```diff
@@ -3228,7 +3228,41 @@ let alert_failed_login_attempts () =
             ~body:stats
       )
 
+let perform ~local_fn ~__context ~host op =
+  let rpc' context hostname (task_opt : API.ref_task option) xml =
+    let open Xmlrpc_client in
+    let transport =
+      SSL
+        ( SSL.make ~verify_cert:(Some Stunnel.pool)
+            ?task_id:(Option.map Ref.string_of task_opt)
+            ()
+        , hostname
+        , !Constants.https_port
+        )
+    in
+    let http =
+      xmlrpc ?task_id:(Option.map Ref.string_of task_opt) ~version:"1.0" "/"
+    in
+    XMLRPC_protocol.rpc ~srcstr:"xapi" ~dststr:"dst_xapi" ~transport ~http xml
+  in
+  let open Message_forwarding in
+  do_op_on_common ~local_fn ~__context ~host op (call_slave_with_session rpc')
```

Comment:
I'm surprised this doesn't create a module circular dependency and dune is happy about this

Reply:
xapi_pool isn't actually used in message_forwarding.ml (we get the desired runtime behaviour because of the functor)